src/test/unit/lambda_tracer/test_lambda_reporter.py

```diff
@@ -28,42 +35,190 @@
+        "mongoConnectionId": "cid",
+        "databaseName": "dname",
+        "mongoRequestId": "rid",
+        "lambda_container_id": "7fa30d38-3aed-4e15-96a7-b53116e2b5fa",
+        "account": "account",
+        "request": '"cmd"',
+        "started": (now - timedelta(seconds=10)).timestamp() * 1000,
+        "transactionId": "transaction-id",
+        "region": "UNKNOWN",
+        "parentId": "parent-id",
+        "info": {"tracer": {"version": "1.1.230"}, "traceId": {"Root": ""}},
+        "token": "token",
+        "commandName": "cname",
+        "ended": now.timestamp() * 1000,
+        "response": '{"code": 200}',
+    }
+
+
+@fixture
+def pymongo_span_metadata(pymongo_span: dict) -> dict:
+    span_copy = copy.deepcopy(pymongo_span)
+    span_copy.pop("request", None)
+    span_copy.pop("response", None)
+    return span_copy
+
+
+@fixture
+def sql_span() -> dict:
+    now = datetime.now()
+    return {
+        "parentId": "1234",
+        "transactionId": "",
+        "values": '["saart"]',
+        "region": "UNKNOWN",
+        "account": "",
+        "query": '"INSERT INTO users (name) VALUES (?)"',
+        "info": {"tracer": {"version": "1.1.230"}, "traceId": {"Root": ""}},
+        "token": "t_10faa5e13e7844aaa1234",
+        "type": "mySql",
+        "started": (now - timedelta(seconds=10)).timestamp() * 1000,
+        "connectionParameters": {
+            "host": "/private/var/folders/qv/w6y030t978518rzpnk1kt0_80000gn/T/pytest-of-nadavgihasi/pytest-29/test_happy_flow0/file.db",
+            "port": 1234,
+            "database": "/private/var/folders/qv/w6y030t978518rzpnk1kt0_80000gn/T/pytest-of-nadavgihasi/pytest-29/test_happy_flow0/file.db",
+            "user": "ng",
+        },
+        "lambda_container_id": "8e86b65e-45b7-46ac-b924-be89113964b7",
+        "id": "e3cac203-50db-43e8-bb6e-07add431edf2",
+        "ended": now.timestamp() * 1000,
+        "response": "very-long-response",
+    }
+
+
+@fixture
+def sql_span_metadata(sql_span: dict) -> dict:
+    span_copy = copy.deepcopy(sql_span)
+    span_copy.pop("query", None)
+    span_copy.pop("values", None)
+    span_copy.pop("response", None)
+    return span_copy
+
+
+def test_create_request_body_default(dummy_span: dict):
     assert _create_request_body([dummy_span], False) == json.dumps([dummy_span])
 
 
-def test_create_request_body_not_effecting_small_events(dummy_span):
+def test_create_request_body_not_effecting_small_events(dummy_span: dict):
     assert _create_request_body([dummy_span], True, 1_000_000) == json.dumps([dummy_span])
 
 
 def test_create_request_body_keep_function_span_and_filter_other_spans(
-    dummy_span, function_end_span
+    dummy_span: dict, function_end_span: dict
 ):
-    expected_result = [dummy_span, dummy_span, dummy_span, function_end_span]
+    input_spans = [dummy_span, dummy_span, dummy_span, function_end_span, function_end_span]
+    expected_result = [function_end_span, function_end_span, dummy_span]
```

Comment:
where are all the dummy_spans? why is there only one left?

Reply:
because the size us limited, I wanted to check that we prioritize the `end_span` before other spans.